validate_hif.py

```diff
@@ -0,0 +1,85 @@
+import json
+import sys
+from collections import defaultdict
+
+# network parameters
+filename = sys.argv[1]
+
+with open(filename) as file:
+    # load JSON file
+    data = json.loads(file.read())
+
+# check that keys do not deviate from the standard field names
+fields = {"network-type", "metadata", "nodes", "edges", "incidences"}
+if not set(data).issubset(fields):
+    raise Exception(f"Acceptable field names are: {", ".join(fields)}\nand the field names are {", ".join(set(data))}")
+
+# incidences are required
+if "incidences" not in data:
+    raise Exception(f"The file must contain an field for incidences.")
+
+# check network type
```

Comment:
add in the comments explanation of each network type. Thus the code of validator can also serve as documentation of the format.

Reply:
Fixed.